### ISSUE TITLE
pip search throws an exception

### ISSUE THREAD

**Issue by platformvaimo** [stevearc/pypicloud#133]
From version 0.5.2 (including 1.0.0rc1), pip search does not work as expected.

Running: `pip search -i http://localhost:6543/pypi/ <query>` will throw an exception:

```
HTTP error 400 while getting http://localhost:6543/pypi/
Exception:
Traceback (most recent call last):
  File "/Users/platform/Envs/pypitest/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Users/platform/Envs/pypitest/lib/python2.7/site-packages/pip/commands/search.py", line 45, in run
    pypi_hits = self.search(query, options)
  File "/Users/platform/Envs/pypitest/lib/python2.7/site-packages/pip/commands/search.py", line 62, in search
    hits = pypi.search({'name': query, 'summary': query}, 'or')
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xmlrpclib.py", line 1240, in __call__
    return self.__send(self.__name, args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xmlrpclib.py", line 1599, in __request
    verbose=self.__verbose
  File "/Users/platform/Envs/pypitest/lib/python2.7/site-packages/pip/download.py", line 776, in request
    response.raise_for_status()
  File "/Users/platform/Envs/pypitest/lib/python2.7/site-packages/pip/_vendor/requests/models.py", line 862, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 400 Client Error: Bad Request for url: http://localhost:6543/pypi/
```
We have tested it on 0.5.0 and 0.5.1 and there it works. Versions 0.5.2, 0.5.3, 0.5.4 and 1.0.0rc1 are all broken.

pip version tested with is 9.0.1

**Comment by stevearc** [comment]
Wow...I don't know how I managed to do this but I really messed up the docs. If you want it to work now, you can omit the trailing slash and that should fix it
`pip search -i http://localhost:6543/pypi <query>`

I'll do a patch release soon that allows the optional trailing slash and updates the docs to be correct.

**Comment by stevearc** [comment]
Released a fix in version [0.5.5](https://pypi.python.org/pypi/pypicloud/0.5.5) that makes the trailing slash url work

**Comment by platformvaimo** [comment]
Great. Thanks! Tested and confirmed that it's fixed in 0.5.5
